sycl/test/esimd/simd_subscript.cpp

```diff
@@ -68,6 +68,7 @@ void test_simd_writable_subscript() SYCL_ESIMD_FUNCTION {
 void test_simd_const_subscript() SYCL_ESIMD_FUNCTION {
   const simd<int, 4> cv = 1;
   int val2 = cv[0]; // returns int instead of simd_view
+  // CHECK: simd_subscript.cpp:72{{.*}}error: expression is not assignable
   cv[1] = 0;        // expected-error {{expression is not assignable}}
```

Comment:
```suggestion
  cv[1] = 0;
```
If I get it right, original test will fail if there are any errors other than the expected one. New testing approach will not detect if this code will emit new errors. I'm not sure if it's import for this test, but you might need add CHECKs that there are no other error messages.
This applies to all ESIMD tests where `-verify` is replaced with `FileCheck`.

Reply:
> If I get it right, original test will fail if there are any errors other than the expected one. New testing approach will not detect if this code will emit new errors. I'm not sure if it's import for this test, but you might need add CHECKs that there are no other error messages.

That is a good point and I'm not sure that it could be achieved reliable enough with `FileCheck`.  I will try to do some experiments. Would it be ok if I do that via separate PR?

Reply:
Yes.

Reply:
> I'm not sure that it could be achieved reliable enough with `FileCheck`.

According to my understanding it's the motivation behind `-verify` feature.

Reply:
> Yes.

Could you please then merge this PR? Its description is intended to be a commit message (it is different from first commit message)

> According to my understanding it's the motivation behind `-verify` feature.

But unfortunately we won't be able to use it for host side - either improvements for `-verify` are needed or we need to find some compromise/alternative solution for our tests.

Reply:
> > Yes.
> Could you please then merge this PR? Its description is intended to be a commit message (it is different from first commit message)

Actually, let me do some experiments first. I totally forgot about `fsycl-is-host` thing I'm adding - probably with its help I can simplify the PR significantly and preserve `-verify` in those tests

Reply:
> Actually, let me do some experiments first. I totally forgot about fsycl-is-host thing I'm adding - probably with its help I can simplify the PR significantly and preserve -verify in those tests

And that won't work, because `-verify` won't work in host mode anymore due to integration footer. However, I've found a way how to check for extra diagnostics using `FileCheck` - I will update the PR and mark it for review again

Reply:
Added 18158a4b7fad405b9becd950bd04dcc704bcebea and 35b6ef9400b308208832e53fc9168167b7f8297b to ensure that extra diagnostics are not emitted